iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/condition_listener.hpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_POSH_POPO_BUILDING_BLOCKS_CONDITION_LISTENER_HPP
+#define IOX_POSH_POPO_BUILDING_BLOCKS_CONDITION_LISTENER_HPP
+
+#include "iceoryx_posh/internal/popo/building_blocks/condition_variable_data.hpp"
+#include "iceoryx_posh/mepoo/memory_info.hpp"
+#include "iceoryx_utils/cxx/helplets.hpp"
+
+namespace iox
+{
+namespace popo
+{
+/// @brief ConditionListener allows one to wait using a shared memory condition variable
+class ConditionListener
+{
+  public:
+    using NotificationVector_t = cxx::vector<cxx::BestFittingType_t<MAX_NUMBER_OF_NOTIFIERS_PER_CONDITION_VARIABLE>,
+                                             MAX_NUMBER_OF_NOTIFIERS_PER_CONDITION_VARIABLE>;
+
+    explicit ConditionListener(ConditionVariableData& condVarData) noexcept;
+    ~ConditionListener() noexcept = default;
+    ConditionListener(const ConditionListener& rhs) = delete;
+    ConditionListener(ConditionListener&& rhs) noexcept = delete;
+    ConditionListener& operator=(const ConditionListener& rhs) = delete;
+    ConditionListener& operator=(ConditionListener&& rhs) noexcept = delete;
```

Comment:
could you add tests to check for the deletion of this, e.g. `EXPECT_FALSE(std::is_move_constructible<ConditionListener>::value);`. It's not to check if the compiler does the right thing but to have a failed test if this is implemented in the future and the developer forgot to write tests for it.